### PR TITLE
Use generated mocks to implement unit tests for pkg/backends 

### PR DESCRIPTION
**Pull request by rramkumar1**
This PR integrates the gce generated mocks into pkg/backends. 

This PR vendors k/pkg/cloudprovider/providers/gce/cloud/mock and there are two raw patches to the vendored code which will need to be upstreamed at some point:
   1. Add a new file called gce_fake.go which exposes a generic fake GCECloud object.
   2. Add new hooks functions to existing mock.go file

This change is [Reviewable](https://reviewable.io/reviews/kubernetes/ingress-gce/303)

### CODE REVIEW

File: pkg/backends/backends_test.go [hunk 1]
```diff
@@ -1041,3 +1000,59 @@ func TestEnsureBackendServiceHealthCheckLink(t *testing.T) {
...
+	}
+
+	for _, tc := range testCases {
+		if comparableGroupPath(tc.igPath) != tc.expected {
+			t.Errorf("expected %s, but got %s", tc.expected, comparableGroupPath(tc.igPath))
+		}
+	}
+}
+
+func TestRetrieveObjectName(t *testing.T) {
```
Comment (nicksardo) [inline pkg/backends/backends_test.go]: Can you move these funcs back to there original position so my PR won't conflict as much?

Comment (rramkumar1) [reply pkg/backends/backends_test.go]: Done.

**Pull request merged**